flexget/plugins/list/movie_list.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import logging
+from collections import MutableSet
+
+from sqlalchemy import Column, Unicode, Integer, ForeignKey, func
+from sqlalchemy.orm import relationship
+from sqlalchemy.sql.elements import and_
+
+from flexget import plugin
+from flexget.db_schema import versioned_base, with_session
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.tools import split_title_year
+
+log = logging.getLogger('movie_list')
+Base = versioned_base('movie_list', 0)
+
+SUPPORTED_IDS = ['imdb_id', 'trakt_movie_id', 'tmdb_id']
+
+
+class MovieListList(Base):
+    __tablename__ = 'movie_list_lists'
+    id = Column(Integer, primary_key=True)
+    name = Column(Unicode, unique=True)
+    movies = relationship('MovieListMovie', backref='list', cascade='all, delete, delete-orphan', lazy='dynamic')
+
+    def to_dict(self):
+        return {
+            'id': self.id,
+            'name': self.name
+        }
+
+
+class MovieListMovie(Base):
+    __tablename__ = 'movie_list_movies'
+    id = Column(Integer, primary_key=True)
+    title = Column(Unicode)
+    year = Column(Integer)
+    list_id = Column(Integer, ForeignKey(MovieListList.id), nullable=False)
+    ids = relationship('MovieListID', backref='movie', cascade='all, delete, delete-orphan')
+
+    def to_entry(self):
+        entry = Entry()
+        entry['title'] = entry['movie_name'] = self.title
+        entry['url'] = 'mock://localhost/movie_list/%d' % self.id
+        if self.year:
+            entry['title'] += ' (%d)' % self.year
+            entry['movie_year'] = self.year
+        for id in self.ids:
+            entry[id.id_name] = id.id_value
+        return entry
+
+    def to_dict(self):
```

Comment:
Honestly, I couldn't tell you if there's any harm in adding all that stuff to entry. I just feel like it has no value there and I have no idea if it can cause issues down the road. I know having to of these representations is a little weird, but each one is used for a different purpose. I feel better knowing that each representation is exactly right for its own purpose instead of manipulating one representations to fit two usages.